mlflow/recipes/utils/wrapped_recipe_model.py

```diff
@@ -16,7 +18,8 @@ def __init__(
     def load_context(self, context):
         self._classifier = mlflow.sklearn.load_model(context.artifacts["model_path"])
 
-    def predict(self, context, model_input):
+    @experimental
+    def predict(self, context, model_input, params: Optional[Dict[str, Any]] = None):
```

Comment:
@serena-ruan Just to confirm, if a user has a preexisting model that was logged before we introduced `params`, users can still load it as pyfunc and call `predict()`? We're not going to break those models by introducing `params` support, right? Have we tested that?

Reply:
From https://github.com/mlflow/mlflow/pull/8563/files#r1247208413, it looks like this might not be the case.

Can we make sure that we have backwards compatibility with preexisting models and add two test cases for this:

1. Using a model logged in MLflow 2.4, load the model as pyfunc via `mlflow,pyfunc.load_model()` and call `predict()` on an input. Verify that it succeeds, even though the model doesn't have `params` support

2. Using a model logged in MLflow 2.4, serve the model with `mlflow models serve --env-manager local` and submit an input via REST call. Verify that the expected output is returned, even though the model doesn't have `params` support

cc @BenWilson2 @harupy

Reply:
Sorry I'm a little bit confused, we didn't change model logging process in this PR, so these two cases are just like 
1. log a model and load it back with pyfunc and call predict without any params. -> I think most of the existing predict function tested are like this, they don't pass params, except some newly added tests in transformers flavor.
2. Same as above but using model serving -> existing tests should cover this as well, and newly added tests make sure passing params with ModelSignature which specified ParamSchema works well.
Am I understanding it correctly?

Reply:
It would be good to have this compatibility check in place just to confirm that there aren't any other nuanced behavior that might be surprising. We've encountered breaking changes like this before that we hadn't anticipated during feature development that were only uncovered when doing this pre-feature-version model logging and loading in current feature created version tests. 
It's definitely worth making sure that we're not going to be surprised by anything when releasing the feature :)